modules/standard/CTypes.chpl

```diff
@@ -140,6 +140,28 @@ module CTypes {
     }
   }
 
+  /*
+    Like ``c_ptr``, but for a pointer to const data. In C, this is equivalent to
+    the type `const eltType*`.
+  */
+  pragma "data class"
+  pragma "no object"
+  pragma "no default functions"
+  pragma "no wide class"
+  pragma "c_ptr class"
+  class c_ptrConst {
```

Comment:
I wouldn't mind the redundancy if you added a `pragma "c_ptrToConst class"` on top of `c_ptr class`. I would also guess that just `c_ptrToConst class` should be relatively easy, but may need compiler modifications beyond what you want for this PR.

Reply:
But maybe add a TODO if you have the redundant pragma?